fuzzing/corpus/corpus.go

```diff
@@ -543,3 +553,77 @@ func (c *Corpus) Flush() error {
 
 	return nil
 }
+
+// PruneSequences removes unnecessary entries from the corpus. It does this by:
+//   - Initialize a blank coverage map tmpMap
+//   - Grab all sequences in the corpus
+//   - Randomize the order
+//   - For each transaction, see whether it adds anything new to tmpMap.
+//     If it does, add the new coverage and continue.
+//     If it doesn't, remove it from the corpus.
+//
+// By doing this, we hope to find a smaller set of txn sequences that still preserves our current coverage.
+// PruneSequences takes a chain.TestChain parameter used to run transactions.
+// It returns an int indicating the number of sequences removed from the corpus, and an error if any occurred.
+func (c *Corpus) PruneSequences(ctx context.Context, chain *chain.TestChain) (int, error) {
+	chainOriginalIndex := uint64(len(chain.CommittedBlocks()))
+	tmpMap := coverage.NewCoverageMaps()
+
+	c.callSequencesLock.Lock()
+	seqs := make([]calls.CallSequence, len(c.mutationTargetSequenceChooser.Choices))
+	for i, seq := range c.mutationTargetSequenceChooser.Choices {
+		seqCloned, err := seq.Data.Clone()
+		if err != nil {
+			c.callSequencesLock.Unlock()
+			return 0, err
+		}
+		seqs[i] = seqCloned
+	}
+	c.callSequencesLock.Unlock()
+	// We don't need to lock during the next part as long as the ordering of Choices doesn't change.
+	// New items could get added in the meantime, but older items won't be touched.
+
+	toRemove := map[int]bool{}
+
+	// Iterate seqs in a random order
+	for _, i := range rand.Perm(len(seqs)) { // TODO should we use a "random provider"?
+		if utils.CheckContextDone(ctx) {
+			return 0, nil
```

Comment:
It doesn't matter too much I guess but shouldn't we return `return toRemove, nil`?